lib/facility.rb

```diff
@@ -1,14 +1,79 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees 
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_writer :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+
+  def initialize(facility_details = {})
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def assign_plate_type(vehicle)
+    if vehicle.antique?
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle?
+      vehicle.plate_type = :ev
+    else
+      vehicle.plate_type = :regular
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+        vehicle.registration_date = Date.today
+        assign_plate_type(vehicle)
+        calculate_fees(vehicle)
+        @registered_vehicles << vehicle
+    else
+      false
+    end
+  end
+
+  def calculate_fees(vehicle)
+    @collected_fees += if vehicle.plate_type == :regular
+      100
+    elsif vehicle.plate_type == :ev
+      200
+    else
+      25
+    end
+  end
+
+  def administer_written_test(registrant)
+    return false unless @services.include?("Written Test")
+    return false unless registrant.permit? == true && registrant.age >= 16
+
+    registrant.license_data[:written] = true
+  end
+
+  def administer_road_test(registrant)
```

Comment:
Good job with these three methods. Because we're checking three conditions though, I would rather just have all three conditions being checked at the same time instead of checking one condition, then the other two. Let's just do it all in one go. so line 61 could like like this: if @services.include?('Written Test') && registrant.age >= 16 && registrant.permit? then just put line 61 inside the if block.